### ISSUE TITLE
Give etag after a put

### ISSUE THREAD

**Issue by sylvainlb** [read-write-web/rww-play#157]
When you PUT a resource to the server, its etag is modified but not communicated to the client. That means that if the client wants to save a new version of the resource, it first needs to do an extra GET on it, although it already has it.

Shouldn't the server add the new etag as a header of the PUT response?

**Comment by bblfish** [comment]
yes, I think I asked a question like that on the ldp mailing list. Mostly these should return the same headers as GET. But would be good to find the HTTP RFC that details that.

**Comment by sylvainlb** [comment]
ok. I'll see if I can fix it myself

**Comment by bblfish** [comment]
Thanks, should be easy.

**Comment by sylvainlb** [comment]
As I understand, PUT, PATCH and POST should return Access-Control-Allow-Origin, Accept-Patch, mimetype, then linkHeaders(ldpr) and the commonHeaders. Rigth?

**Comment by bblfish** [comment]
I don't think in this case the order is important

**Comment by sylvainlb** [comment]
Ok, but I was checking that my list was correct.

**Comment by bblfish** [comment]
POST should definitely return the headers that the created document should return on a GET. For PUT and PATCH they should return the headers of the document that were changed. And yes, that seems to include Access-Control stuff and what patch formats the resource accepts.

**Comment by sylvainlb** [comment]
Ok, I couldn't do it so far. I need an AuthResult to call the header functions, and all I have is an IdResult. I'm not sure if I could get the first from the latter or if I should call fullAuthInfo. I'll try to keep looking, but if you have a clue on how to get that AuthResult, it's welcome.

**Comment by bblfish** [comment]
The problem is that the [`put` function](https://github.com/read-write-web/rww-play/blob/901a5250e9e62d101265cdb9a57c36b7f6623bd3/app/rww/play/ResourceManager.scala#L116-L116) returns an `IdResult[Rdf#URI]`. Further down the problem is that the  `putLDPR` script returns returns Unit. You can see this by comparing with [getLDPR](https://github.com/read-write-web/rww-play/blob/bfa76510b44cdcc7acb003ac004809d89f4632f0/app/rww/ldp/LDPCommand.scala#L140-L139), which returns an `Rdf#Graph`.

I don't think it really makes much sense to have commands that just return graphs. The metadata about the graphs, is so often important that I think the command structure needs to be simplfied for that.

Here the simple answer for the moment is to follow the put by a getMeta, and then use that information.
It would actually be easy to optimize the code so that this is very cheap. There are loads of examples of that in the tests.
